src/services/document-input-service.js

```diff
@@ -41,19 +41,40 @@ class DocumentInputService {
 
   async getDocumentInputsCreatedByUser(userId) {
     const documentInputs = await this.documentInputStore.getDocumentInputsCreatedByUser(userId);
-    return documentInputs.sort(by(input => input.createdOn, 'desc'));
+    return documentInputs
+      .map((documentInput, index) => {
+        return {
+          ...documentInput,
+          inputOrder: index + 1
+        };
+      })
+      .sort(by(input => input.createdOn, 'desc'));
```

Comment:
Maybe that's very theoretical, but now we set the order number before we actually sort (given the db returns the inputs in ascending order by default). It's a little bit irritating when reading the code. It would be nicer to sort before setting the `inputOrder`, e.g.:

~~~
    return documentInputs
      .sort(by(input => input.createdOn, 'desc'))
      .map((documentInput, index, allInputs) => {
        return {
          ...documentInput,
          inputOrder: allInputs - index
        };
      });
~~~

Reply:
I've undone the changes on the server side, as I agree it's confusing and actually not the right place for this.
I determine the order on the client side, in the one single place that needs it.